src/client_networking.rs

```diff
@@ -55,49 +55,23 @@ impl From<StrictError> for ConnectionError {
 }
 
 
-// I'd change the declaration to: request_table(table_name: &str, server_address: &str)
-// Agree with name => table_name but this gets a csv. Should be called download_csv, though, to be consistent with server()
 pub fn request_csv(name: &str, address: &str) -> Result<StrictTable, ConnectionError> {
-    // Validating the server_address somehow and providing a descriptive error would probably be a good idea
 
     let mut connection: TcpStream = TcpStream::connect(address)?;
-    
-    // The writing to the connection here is the same code as elsewhere
-    // perhaps you want to encapsulate the writing/communications themselves somehow. 
-    // E.g. say you change the server to accept json or another format, 
-    // you'd only want to make the formatting change in one place.
-    // It doesn't have to be a class but some intermideary between this request_table function
-    // and the connection that would surface say at least two methods: 
-    // send_request(method: requesting|sending|other, params: &str), 
-    //      where the method param would be an enum allowing you to 
-    //      switch between the appearent methods however that's best done in rust
-    //      and the other would be the params (like name) currently going into the formatted string.
-    //      This method could then also handle parsing the data from the server  (i.e. the read in the loop below)
-    //      and casting the bytes to a string
-    // and 
-    // send_confirmation(), which would send the "OK"
-
-    //      This is this way for the print statement for debugging reasons. Will be this in future:
-    //        connection.write(format!("admin|admin|Requesting|{}", name).as_bytes())?;
 
     match connection.write(format!("admin|admin|Requesting|{}", name).as_bytes()) {
         Ok(n) => println!("Wrote request as {n} bytes"),
         Err(e) => {return Err(ConnectionError::Io(e));},
     };
 
     let mut buffer: [u8; BUFFER_SIZE] = [0; BUFFER_SIZE];
-    // What the hell are you looping over here? Hahah, rust looks weird coming from js 😂 but very cool
-    // This loop is needed to read large data but would not work as is. That problem is solved in server()
-    // but not here yet
+    
     loop {
         match connection.read(&mut buffer) {
             Ok(_) => break,
             Err(e) => {return Err(ConnectionError::Io(e));}        }
     }
 
-    // This would then be the result of the call to send_request, 
-    // i'd maybe rename it "response" because its not necissarily csv which you clearly know ofc
-    // For the foreseeable future this will remain a csv and I like descriptive names.
```

Comment:
Right yeah, no, I definitely agree with that. I was more alluding to that while the string `No such table` is I guess strictly in csv format, I don't think about a payload being of a given type until it has passed some validation. It felt a bit odd to me that checking that a csv payload is a given string, or in this case really an error message.

That could be another benefit of abstracting away the communications somehow, the communication interface could handle error checking and raise actual errors in case of a 404 such as this, if you were to implement some standard of reporting errors such as the HTTP codes or something similar.

Reply:
Hmmm, I don't know how to throw an error through a TcpStream other than just sending some data that will be read on the other end and interpreted. "No such table" seemed pretty descriptive in the case of a key error. The validation of the csv is handled in the next line. The question mark will throw a ConnectionError::StrictError if the received csv is not strict (which shouldn't happen since the server also validates but you can never be too strict).

Reply:
No yeah, the error itself is good (for now at least), that's not what I meant! This comment was nitpicky from the start 😅 It was just to get at the fact that the variable `csv` could potentially contain the error response "No such table" and was not guaranteed to be a csv string. Again, real nitpicky. I think we might both benefit from me throwing up a bit of sheep code (pseudo code er sauðakóði á íslensku 😅) I'll do that in a separate comment.

Reply:
Sauðakóði er geggjað orð

Reply:
Ég geri mér enga grein fyrir því hversu mikið þú veist af svona random dóti á íslensku náttúrulega

Reply:
Sirka ekkert

Reply:
Hahah, datt það í hug, I'll drop nuggets jafnóðum og þeir hvarfla að mér. 

Annar 
"SQL" 
Oft borið fram á ensku:
"Ess - Queue - Ell"
Eða
"See - Qwuhl"

En íslenska "Skúli" stundum notað af proper nöllum 😅